catalogd/internal/storage/localdir_test.go

```diff
@@ -13,219 +10,498 @@ import (
+		t.Fatal("failed to store test catalog")
+	}
+	testServer := httptest.NewServer(store.StorageServerHandler())
+
+	testCases := []struct {
+		name               string
+		setupStore         func() (*httptest.Server, error)
+		queryParams        string
+		expectedStatusCode int
+		expectedContent    string
+	}{
+		{
+			name:               "valid query with package schema",
+			queryParams:        "?schema=olm.package",
+			expectedStatusCode: http.StatusOK,
+			expectedContent:    `{"defaultChannel":"preview_test","name":"webhook_operator_test","schema":"olm.package"}`,
+		},
+		{
+			name:               "valid query with schema and name combination",
+			queryParams:        "?schema=olm.package&name=webhook_operator_test",
+			expectedStatusCode: http.StatusOK,
+			expectedContent:    `{"defaultChannel":"preview_test","name":"webhook_operator_test","schema":"olm.package"}`,
+		},
+		{
+			name:               "valid query with channel schema and package name combination",
+			queryParams:        "?schema=olm.channel&package=webhook_operator_test",
+			expectedStatusCode: http.StatusOK,
+			expectedContent:    `{"entries":[{"name":"bundle.v0.0.1"}],"name":"preview_test","package":"webhook_operator_test","schema":"olm.channel"}`,
+		},
+		{
+			name:               "query with all meta fields",
+			queryParams:        "?schema=olm.bundle&package=webhook_operator_test&name=bundle.v0.0.1",
+			expectedStatusCode: http.StatusOK,
+			expectedContent:    `{"image":"quaydock.io/namespace/bundle:0.0.3","name":"bundle.v0.0.1","package":"webhook_operator_test","properties":[{"type":"olm.bundle.object","value":{"data":"dW5pbXBvcnRhbnQK"}},{"type":"some.other","value":{"data":"arbitrary-info"}}],"relatedImages":[{"image":"testimage:latest","name":"test"}],"schema":"olm.bundle"}`,
+		},
+		{
+			name:               "valid query for package schema for a package that does not exist",
+			queryParams:        "?schema=olm.package&name=not-present",
+			expectedStatusCode: http.StatusOK,
+			expectedContent:    "",
+		},
+		{
+			name:               "valid query with package and name",
+			queryParams:        "?package=webhook_operator_test&name=bundle.v0.0.1",
+			expectedStatusCode: http.StatusOK,
+			expectedContent:    `{"image":"quaydock.io/namespace/bundle:0.0.3","name":"bundle.v0.0.1","package":"webhook_operator_test","properties":[{"type":"olm.bundle.object","value":{"data":"dW5pbXBvcnRhbnQK"}},{"type":"some.other","value":{"data":"arbitrary-info"}}],"relatedImages":[{"image":"testimage:latest","name":"test"}],"schema":"olm.bundle"}`,
+		},
+		{
+			name:               "query with non-existent schema",
+			queryParams:        "?schema=non_existent_schema",
+			expectedStatusCode: http.StatusOK,
+			expectedContent:    "",
+		},
+		{
+			name:               "cached response with If-Modified-Since",
+			queryParams:        "?schema=olm.package",
+			expectedStatusCode: http.StatusNotModified,
+			expectedContent:    "",
+		},
```

Comment:
Add another test cases with an unknown parameter, expected to return BadRequest

Reply:
And another when we get duplicate parameters? (i.e. 'query?package=foo&package=foo')